src/arch/dotproductavx.cpp

```diff
@@ -16,8 +16,9 @@
 ///////////////////////////////////////////////////////////////////////
 
 #if !defined(__AVX__)
```

Comment:
How do you make sure that this code is compiled with AVX compiler options for x86 targets?

Reply:
__AVX__ is defined by the compiler (gcc) if -mavx is passed to it. On *nix the gs configure script sets flags which pass down from the top-level Makefile (which contains conditionals etc) through the platform-independent makefiles to our tesseract.mak file.

On Windows, our top-level msvc.mak file predefines the appropriate things in the CFLAGS.

Getting flags through the make system is relatively simple. Selecting between different sets of files to build is trickier.

If you're really interested, the whole nightmarish complexity of the changes to the Ghostscript build can be seen within the commit here: https://git.ghostscript.com/?p=user/robin/ghostpdl.git;a=commitdiff;h=e8fb4eb5f7e7d6514d7969e94a909654fe690fdf

(at least until I revise the commit and git garbage collects it)

Reply:
Those files are special because they require compiler flags which must not be used for other files. Example: if you compile all code with `-mavx2`, the resulting binary will no longer work on a host without AVX2.

You are using a normal `autoconf` build as far as I see. Then I suggest to build Leptonica and Tesseract as subprojects with their own `autoconf` build process and simply use the resulting libraries for your build.

What is the use of duplicating build rules?

Reply:
On *some* platforms we use an autoconf based build, but not on all. Windows is the classic example, but there are others.

Yes, I know about -mavx2 only being used for the avx2 files etc. My code does that correctly.